### PR TITLE
Bug 1810114 - Move marketingNotificationAllowed metric to FenixApplication

### PR DESCRIPTION
**Pull request by rahulsainani**
– Now that [Notification pre permission prompt](https://bugzilla.mozilla.org/show_bug.cgi?id=1808874) is merged,
this moves `marketingNotificationAllowed`metric as that was called from `ensureMarketingChannelExisits` which is now behind an experiment, so now we get this metric for all app sessions.
– Extract common helper functions to `NotificationManagerCompat` and add tests.

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

File: app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        tested = mockk(relaxed = true)
+
+        mockkStatic("org.mozilla.fenix.ext.NotificationManagerCompatKt")
+    }
+
+    @Test
+    fun `areNotificationsEnabledSafe returns false when areNotificationsEnabled throws an exception`() {
+        every { tested.areNotificationsEnabled() } throws RuntimeException()
+
+        assertEquals(false, tested.areNotificationsEnabledSafe())
```
Comment (t-p-white) [inline app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt:34]: maybe 
```suggestion
        assertFalse(tested.areNotificationsEnabledSafe())
```
nit: think the assertBoolean wrapper is preferable to assertEquals(Boolean... ? Would apply to the other tests in this file.

File: app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    @Before
+    fun setup() {
+        tested = mockk(relaxed = true)
+
+        mockkStatic("org.mozilla.fenix.ext.NotificationManagerCompatKt")
+    }
+
+    @Test
+    fun `areNotificationsEnabledSafe returns false when areNotificationsEnabled throws an exception`() {
```
Comment (t-p-white) [inline app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt:31]: Nice tests 👍 
Generally I think the [GIVEN, WHEN, THEN] or [WHEN, THEN] format is preferred for test names, examples: https://github.com/mozilla-mobile/fenix/blob/main/app/src/test/java/org/mozilla/fenix/gleanplumb/DefaultMessageControllerTest.kt & https://github.com/mozilla-mobile/fenix/blob/main/app/src/test/java/org/mozilla/fenix/browser/BrowserFragmentTest.kt

File: app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    @Test
+    fun `areNotificationsEnabledSafe returns true when areNotificationsEnabled returns true`() {
+        every { tested.areNotificationsEnabled() } returns true
+
+        assertEquals(true, tested.areNotificationsEnabledSafe())
+    }
+
+    @Test
+    fun `isNotificationChannelEnabled returns true when getNotificationChannelCompat returns a channel and areNotificationsEnabled returns true`() {
```
Comment (t-p-white) [inline app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt:52]: should we also have a test for when the channel importance is NotificationManagerCompat.IMPORTANCE_NONE and a test for when the build version is < or = to Build.VERSION_CODES.O ?

**Review by t-p-white — CHANGES_REQUESTED** [review]
Looks good 👍 Just a couple of comments around tests

File: app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt (hunk 2 above)
Comment (rahulsainani) [reply app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt]: Thanks for the reference 👌  updated to follow the same pattern

File: app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt (hunk 3 above)
Comment (rahulsainani) [reply app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt]: Good point, updated to have `IMPORTANCE_NONE` test case 👍 
I was checking how to fake a different OS version, roboelectric has some ways to config the sdk used, which would run the tests in a different environment/config. Having an abstraction like BuildVersionProvider and fake in test might work but seemed a bit much for just this use case 😅  do you know have any ideas how to go about this?

Comment (rahulsainani) [reply app/src/test/java/org/mozilla/fenix/ext/NotificationManagerCompatTest.kt]: Added the test using `ReflectionHelpers` and a follow up ticket for `BuildVersionProvider` we discussed to improve this --> https://bugzilla.mozilla.org/show_bug.cgi?id=1810179

**Review by t-p-white — APPROVED** [review]
lgtm 👍

**Pull request merged**